packages/google_maps_flutter/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -250,7 +258,23 @@ class _GoogleMapState extends State<GoogleMap> {
 
   @override
   Widget build(BuildContext context) {
-    return GoogleMapsFlutterPlatform.instance.buildView(
+    final GoogleMapsFlutterPlatform platform =
+        GoogleMapsFlutterPlatform.instance;
+    if (platform is MethodChannelGoogleMapsFlutter) {
```

Comment:
Why can't the new method be called unconditionally? The delegation to the old method was supposed to be handled by a default implementation in the platform interface.

Reply:
The default implementation (`MethodChannelGoogleMapsFlutter`) does call the old method if the platform is not android or hybrid composition is turned off. However, web switches to another platform instance: https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_flutter_web.dart#L13. 

So in this case, we would need to call the other method, right?

Reply:
Sorry, that was poorly phrased; by "default implementation" I mean the code that's part of the base class that all implementations extend from. Having a fallback implementation there is what makes this overall change harmless even for federated implementations we don't even know about.

Reply:
Ah ok. So [buildViewWithTextDirection](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/method_channel/method_channel_google_maps_flutter.dart#L464) was only added to `MethodChannelGoogleMapsFlutter` and not to [GoogleMapsFlutterPlatform](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart#L26). This is same for the flag `useAndroidViewSurface`. This was done because this was an Android only requirement. 

Were you thinking that `buildViewWithTextDirection` should have been added to `GoogleMapsFlutterPlatform` as well? I think that makes more sense and could be done without a breaking change.

Reply:
I made the change in https://github.com/flutter/plugins/pull/4121. How would we make this temporary? Would we just make a breaking change for the platform_interface to make the transition to the new method?

Reply:
Maybe I misunderstood about it being temporary. What's the long-term plan for these methods? Are we keeping both forever?

Reply:
For this method to only be temporary, the platform_interface would have to be broken and bumped to version 3.0. This is because any implementation would need to add the additional parameter to their method. However, I believe the policy is to avoid breaking changes to the platform_interface.

An alternative is to have two methods, and to call `buildViewWithTextDirection` by default in `google_maps_flutter`. But this would still break any implementation that doesn't implement `buildViewWithTextDirection` as the method would throw an `UnimplementedError()`. 

This PR essentially makes `buildViewWithTextDirection` unique to `MethodChannelPlatform`. The `GoogleMap` widget then calls it iff it is `MethodChannelPlatform`. I believe this avoids any breaking changes and any of the [implementations](https://pub.dev/packages?q=dependency%3Agoogle_maps_flutter_platform_interface) would still work with the plugin.

Reply:
> An alternative is to have two methods, and to call `buildViewWithTextDirection` by default in `google_maps_flutter`. But this would still break any implementation that doesn't implement `buildViewWithTextDirection` as the method would throw an `UnimplementedError()`.

That's why I had recommended having the method not throw UnimplementedError, but instead calling the old method.

But my question wasn't about implemented, but the ideal goal: do we want this new parameter forever, or is the need for this parameter temporary? I had thought it was the latter, but again, I may have misunderstood.

Reply:
Ah, you wanted that done at the interface level and not the implementation level. That makes since. 

This parameter is required for Android hybrid composition unless we don't want the user to be able to change the text direction. I haven't seen a way around it yet that doesn't involve framework changes. My intention is that it's a permanent parameter.

Reply:
Okay, sorry for the confusion on my part. In that case yes, I think we should update the platform interface to have a default implementation that calls the old method with a default value. Then implementations (those that want a different behavior than that default) can override both, but nobody will be broken and the details of who implements what won't bleed into this package.

Reply:
That works for me! I updated https://github.com/flutter/plugins/pull/4121 to reflect this.